### PR TITLE
Secure grpc start clean code

### PR DESCRIPTION
**Pull request by hzxuzhonghu**

### CODE REVIEW

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -1036,90 +1035,79 @@ func (s *Server) initGrpcServer() {
...
 	certDir := pilot.CertDir
 	if certDir == "" {
-		certDir = PilotCertDir // /etc/certs
+		certDir = PilotCertDir
 	}
-	if !strings.HasSuffix(certDir, "/") {
-		certDir = certDir + "/"
-	}
-
-	for i := 0; i < 30; i++ {
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go]: Do not like this style in pilot.

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -980,9 +980,8 @@ func (s *Server) initDiscoveryService(args *PilotArgs) error {
 				log.Warna(err)
 			}
 		}()
-		if len(args.DiscoveryOptions.SecureGrpcAddr) > 0 {
```
Comment (rshriram) [inline pilot/pkg/bootstrap/server.go]: what about this check?

Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/server.go]: The length must be >0, because of L964~L967

**Comment by incfly** [comment]
I like the simplification of this PR.

However, I'm wondering the background we introduce retry in the first place. Could it be due to the racing condition? Namely Citadel hasn't prepared key cert for Pilot (or k8s does not mount secrets into Pilot volume yet) when Pilot tries to use the key & cert to start grpc server.

Might be helpful to dig some archaeology a little bit :) 

/cc @myidpt

**Comment by hzxuzhonghu** [comment]
>Namely Citadel hasn't prepared key cert for Pilot 

Fail this sounds reasonable.

>(or k8s does not mount secrets into Pilot volume yet) when Pilot tries to use the key & cert to start grpc server.

It is impossible. K8S/docker  guarantee secret must be mounted before run the process. If secret is not exist, the pod would not run.

**Comment by hzxuzhonghu** [comment]
/test istio-unit-tests

unrelated ut `TestFull` flake

**Pull request merged**